test/object-store/transaction_log_parsing.cpp

```diff
@@ -2119,70 +1928,7 @@ TEMPLATE_TEST_CASE("DeepChangeChecker", "[notifications]", ListOfObjects, ListOf
         REQUIRE_FALSE(_impl::DeepChangeChecker(info, *table, related_tables, key_path_array_test_type, true)(0));
     }
 
-    SECTION("link chains are tracked") {
-        r->begin_transaction();
-        for (int i = 0; i < 10; ++i)
-            objects.push_back(table->create_object());
-        for (int i = 0; i < 19; ++i)
-            objects[i].set(cols[1], objects[i + 1].get_key());
-        r->commit_transaction();
-
-        auto info = track_changes([&] {
-            objects[19].set(cols[0], -1);
-        });
-
-        SECTION("without filter - up to 4 levels deep") {
-            _impl::DeepChangeChecker checker(info, *table, related_tables, key_path_array_empty, false);
-            CHECK(checker(19));
-            CHECK(checker(18));
-            CHECK(checker(16));
-            CHECK_FALSE(checker(15));
-
-            // Check in other orders to make sure that the caching doesn't effect
-            // the results
-            _impl::DeepChangeChecker checker2(info, *table, related_tables, key_path_array_empty, false);
-            CHECK_FALSE(checker2(15));
-            CHECK(checker2(16));
-            CHECK(checker2(18));
-            CHECK(checker2(19));
-
-            _impl::DeepChangeChecker checker3(info, *table, related_tables, key_path_array_empty, false);
-            CHECK(checker3(16));
-            CHECK_FALSE(checker3(15));
-            CHECK(checker3(18));
-            CHECK(checker3(19));
-        }
-
-        SECTION("with filter - more than 4 levels deep") {
-            KeyPath key_path_five_levels = {pair_link, pair_link, pair_link, pair_link, pair_int};
-            KeyPathArray key_path_array_five_levels = {key_path_five_levels};
-
-            _impl::CollectionKeyPathChangeChecker checker(info, *table, related_tables, key_path_array_five_levels,
-                                                          false);
-            CHECK_THROWS(checker(19));
-            CHECK_THROWS(checker(18));
```

Comment:
A functional change in the new test code is that calls like these for an object which has null links along the path to check for no longer throws an invalid object exception. I'm not sure why we would want to throw in that case so I changed it not to.